t/ui/21-admin-needles.t

```diff
@@ -111,6 +112,59 @@ like(
     "redirected to right module too"
 );
 
+subtest 'dereference symlink when displaying needles info' => sub {
+    my $real_needle_dir    = getcwd . '/t/data/openqa/share/tests/opensuse';
+    my $symlink_needle_dir = getcwd . '/t/data/openqa/share/tests/test_symlink_dir';
+    symlink($real_needle_dir, $symlink_needle_dir) or die "Cannot make symlink $!";
+    my $needle_dir            = $schema->resultset('NeedleDirs');
+    my $symlink_needle_dir_id = $needle_dir->create(
+        {
+            path => $symlink_needle_dir . '/needles',
+            name => 'symlink_needle_dir'
+        })->id;
+    my $real_needle_dir_id = $needle_dir->create(
+        {
+            path => $real_needle_dir . '/needles',
+            name => 'real_needle_dir'
+        })->id;
```

Comment:
you can do a for loop to apply the same for both dirs.

Reply:
I think it is ok. The assignment is to different variables and the 2 hash entries are different.